clang/include/clang/Basic/Attr.td

```diff
@@ -1993,6 +1993,26 @@ def CHERISubobjectBoundsUseRemainingSize : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+
+// Attributes for __uintcap_t/__intcap_t to indicate whether
+// they carry provenance or not (to avoid the LHS carries
+// provenance rule that we were using before).
+def CHERINoProvenance : DeclOrTypeAttr {
+  let Spellings = [GNU<"cheri_no_provenance">,
+                   CXX11<"cheri","no_provenance">,
+                   C2x<"cheri", "no_provenance">];
+  let Documentation = [Undocumented]; // FIXME
+}
+// The opposite of the previous attribute
+// XXX: not sure this is particularly useful
```

Comment:
I agree this seems unlikely to be useful in practice.

Reply:
Given the code isn't consulting this attribute beyond creating it, I think this should be dropped. It can be added later if needed.